ci/README.md

```diff
@@ -10,6 +10,14 @@ In order to ensure that the needed packages are generally available without buil
 
 Run [`update-pinned.sh`](./update-pinned.sh) to update it.
 
+## GitHub specific code
+
+Some of the code is specific to GitHub.
+This code is currently spread out over multiple places and written in both Bash and JavaScript.
+The goal is to eventually have all GitHub specific code in `ci/github-script` and written in JavaScript via `actions/github-script`.
+A lot of code has already been migrated, but some Bash code still remains.
+New code must not be introduced as Bash, but as JavaScript instead.
```

Comment:
I hope we are still ok with "trivial" (e.g. one-liner) steps being bash, e.g.:

- https://github.com/NixOS/nixpkgs/blob/127dea8a131f8706e2f0cf9997f9d0d2820a48c5/.github/workflows/backport.yml#L47

Or direct command invocation that is unrelated to github APIs, e.g.:
- https://github.com/NixOS/nixpkgs/blob/127dea8a131f8706e2f0cf9997f9d0d2820a48c5/.github/workflows/check.yml#L42
- https://github.com/NixOS/nixpkgs/blob/127dea8a131f8706e2f0cf9997f9d0d2820a48c5/.github/workflows/build.yml#L68
- https://github.com/NixOS/nixpkgs/blob/127dea8a131f8706e2f0cf9997f9d0d2820a48c5/.github/workflows/check.yml#L93

I'm not sure how to convey this nuance in the wording, but my interpretation of your current wording would be that the above examples are not permitted.